### PR TITLE
circular linked list-Java

### LINKED ISSUE #1285: circular linked list in Java
I'd like to work on this

### PR DESCRIPTION
**Pull request by dibyaaaaax**
Fixes #1285 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [NA] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Java Implementation of Circular Linked list
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Circular_Linked_List/Circular_Linked_List.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Circular_Linked_List/Circular_Linked_List.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
+class node{
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add a space before ``{``.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    int data;
+    node next;
+    
+    node(int data, node next){
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Space before ``{``.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        this.data=data;
+        this.next=next;
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Please follow coding conventions throughout the code.
https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html

File: Circular_Linked_List/Circular_Linked_List.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    }
+}
+
+class cirLinkedList{
+    node head;
+    node tail; //tail is the node that points to the head. created to complete some operations like append in O(1) instead of O(n).
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add a comment before the line that you want to explain, and add a blank line before the comment line.
If the comment is more than 100 characters, split it into multiple lines.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    int data;
+    node next;
+    
+    node(int data, node next){
+        this.data=data;
+        this.next=next;
+    }
+}
+
+class cirLinkedList{
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Space before ``{``.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    node head;
+    node tail; //tail is the node that points to the head. created to complete some operations like append in O(1) instead of O(n).
+    
+
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Remove extra lines like these.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        this.next=next;
+    }
+}
+
+class cirLinkedList{
+    node head;
+    node tail; //tail is the node that points to the head. created to complete some operations like append in O(1) instead of O(n).
+    
+    
+    //function to prepend to the list as per README, i.e, to add a node in front of the head
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add a space after **//** in each comment.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    void append(int num) {
+        node temp=new node(num, null);
+        
+        //if the list is empty, the new node becomes the head (first element of the list)
+        if (head==null) {
+            head=temp;
+            tail=head;
+            tail.next=head;
+        }
+        //append new node before head, declare it as the new tail
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add one blank line before this.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+        cll.append(9);  // 2 -> 9
+        cll.prepend(1);  // 1 -> 2 -> 9
+        cll.insert_after_val(11, 2);  // 1 -> 2 -> 11 -> 9
+        cll.displayList();
+        cll.deleteHead();  // 2 -> 11 -> 9
+        cll.displayList();
+        
+    }
+
+}
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add a sample input-output at the end of file.

Comment (dibyaaaaax) [reply Circular_Linked_List/Circular_Linked_List.java]: I have made the necessary changes. Please review

File: Circular_Linked_List/Circular_Linked_List.java [hunk 10]
```diff
@@ -0,0 +1,178 @@
+import java.util.Scanner;
+
+
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Remove

File: Circular_Linked_List/Circular_Linked_List.java [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+    int data;
+    node next;
+    
+    node(int data, node next) {
+        this.data = data;
+        this.next = next;
+    }
+}
+
+
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Remove

File: Circular_Linked_List/Circular_Linked_List.java [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+        while (n > 0) {
+        	n--;
+        	int t = input.nextInt();
+        	if (t == 1) {
+        		cll.append(input.nextInt());
+        	}
+        	if (t == 2) {
+        		cll.prepend(input.nextInt());
+        	}
+        	if ( t == 3) {
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Remove space before 't'

File: Circular_Linked_List/Circular_Linked_List.java [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+        		cll.insert_after_val(input.nextInt(), input.nextInt());
+        	}
+        	if  (t == 4) {
+        		cll.deleteHead();
+        	}
+        	if (t == 5) {
+        		cll.displayList();
+        	}
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Fix indentation in this line.

File: Circular_Linked_List/Circular_Linked_List.java [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+            pointer = pointer.next;
+        }
+        System.out.println();
+        return;
+    }
+}
+
+public class Circular_Linked_List {
+    public static void main(String[] args) {
+        cirLinkedList cll=new cirLinkedList();
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Add space around **=**

File: Circular_Linked_List/Circular_Linked_List.java [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+        Scanner input = new Scanner(System.in);
+        System.out.println("Number of operations: ");
+        int n = input.nextInt();
+       
+        // perform n operations
+        while (n > 0) {
```
Comment (marwahmanbir) [inline Circular_Linked_List/Circular_Linked_List.java]: Fix indentation in whole while loop.

**Pull request merged**